src/api/feed-discovery/src/lib.js

```diff
@@ -0,0 +1,104 @@
+const got = require('got');
+const cheerio = require('cheerio');
+const validUrl = require('valid-url');
+const createError = require('http-errors');
+
+// Helper function to return the feed url of a given blog url
+function getFeedUrls(document) {
+  // Load the html document
+  const $ = cheerio.load(document);
+  // Extract the Cheerio object of rss link elements
+  const rss = $('link[type="application/rss+xml"]').clone();
```

Comment:
Question: why do you `.clone()` in all cases, and not just use the object returned?  Also, there's a lot of code duplication in this function.

I think we should simplify a bit.  For starters, we could combine how you're collecting all these links, since each one is just a similar selector:

```js
const selectors = [
  'link[type="application/rss+xml"]',
  'link[type="application/atom+xml"]',
  ...
].join(',');
const links = $(selectors);
```
Then you can do your iteration once:

```js
const feedUrls = [];

if(links) {
  links.each((i, elem) => {
    const url = elem.attr('href');
    if(href) {
      feedUrls.push(url);
    }
  });
}
```

Reply:
@humphd : done! Thanks for the feedbacks. I am still learning how Cheerio works so this has been at the back of my mind.